.buildkite/publish/test-docker.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
```

Comment:
Could you add some comment blocks to these files to explain at a high level what's happening? I followed the logic of them until this one, which looks like it's using bash to define a run-time addition of an extra buildkite job? If I'm confused now, I expect I'll be extra confused if I have to come troubleshoot something here when a release fails - so any extra explanation/breadcrumbs would be awesome.